src/storage/config.rs

```diff
@@ -54,9 +54,11 @@ impl Default for Config {
 
 impl Config {
     pub fn validate(&mut self) -> Result<(), Box<Error>> {
-        if self.data_dir != DEFAULT_DATA_DIR {
-            self.data_dir = try!(config::canonicalize_path(&self.data_dir))
+        if self.data_dir.is_empty() {
```

Comment:
Yes. But storage use config to create an EngineRocksdb. See https://github.com/pingcap/tikv/blob/ef23163293faedf46e9b65f1991e572d72230a0c/src/storage/mod.rs#L436.